src/main/telemetry/crsf.c

```diff
@@ -337,17 +337,10 @@ void crsfFrameDeviceInfo(sbuf_t *dst) {
 
 static uint16_t getRunLength(const void *start, const void *end)
 {
-    uint8_t *cursor = (uint8_t*)start;
-    uint8_t c = *cursor;
-    size_t runLength = 0;
-    for (; cursor != end; cursor++) {
-        if (*cursor == c) {
-            runLength++;
-        } else {
-            break;
-        }
-    }
-    return runLength;
+    const uint8_t *p = start;
+
+    while ((void *) start == (void *) p && p != end) p++;
```

Comment:
`while (*(uint8_t*)start == *p && p != end) p++;`

Or even better, change parameters to `uint8_t *` and avoid all casting (RLE works on bytes, so it makes more sense to pass byte pointers. `void*` is usually used for binary data block)

Reply:
But then you also have to change function declarion? (And all calls to this function)
`static uint16_t getRunLenght(const void *start, const void* end){ ... }`
===>
`static uint16_t getRunLenght(const uint8_t *start, const uint8_t *end){ ... }`

Reply:
There is no declaration, only definition.
And conversion between `type *` and `void *` is implicit, so there is no need to change call site (in this case)